solutions/usaco-915.mdx

```diff
@@ -73,4 +73,36 @@ class Main {
 
 </JavaSection>
 
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+int main(int argc, char **argv) {
+  string prog_name = "herding";
+  freopen((prog_name + ".in").c_str(), "r", stdin);
+  freopen((prog_name + ".out").c_str(), "w", stdout);
```

Comment:
just `freopen("herding.in", "r", stdin)` and `freopen("herding.out", "w", stdout)` would be fine